client/my-sites/checkout/src/hooks/use-tos-foldable-card.tsx

```diff
@@ -1,7 +1,29 @@
-import { hasCheckoutVersion } from '@automattic/wpcom-checkout';
+import isAkismetCheckout from 'calypso/lib/akismet/is-akismet-checkout';
+import { useExperiment } from 'calypso/lib/explat';
+import isJetpackCheckout from 'calypso/lib/jetpack/is-jetpack-checkout';
+import { useSelector } from 'calypso/state';
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
 export function useToSFoldableCard(): boolean {
-	if ( hasCheckoutVersion( 'tos-foldable-card' ) ) {
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+
+	const isJetpackNotAtomic = useSelector(
+		( state ) => isJetpackSite( state, siteId ) && ! isSiteAutomatedTransfer( state, siteId )
+	);
+
+	const isWPcomCheckout = ! isJetpackCheckout() && ! isAkismetCheckout() && ! isJetpackNotAtomic;
+	const [ isLoadingExperimentAssignment, experimentAssignment ] = useExperiment(
+		'wp_web_checkout_tos_foldable_card_v1',
+		{ isEligible: isWPcomCheckout }
+	);
+
+	if ( ! isLoadingExperimentAssignment ) {
+		return false;
```

Comment:
If we are not loading we are returning false? Doesn't make sense to me.

Also we shouldn't display any experience if we are loading.

Reply:
Updated the useToSFoldableCard hook to return `'loading' | 'treatment' | 'control'` which is then used to modify the `checkoutLoadingConditions` array as detailed by @sirbrillig . The treatment string is then used to show the ToS Foldable Card treatment, whereas the default experience will show for the control group. @jessie-ross please let me know if there are any further issues 👍